src/feditest/protocols/activitypub/__init__.py

```diff
@@ -28,19 +28,23 @@ def __init__(self, uri: str):
         self._json : Any | None = None
 
 
-    def _ensure_fetched(self):
+    def _ensure_fetched(self) -> None:
         """
         Make sure the uri has been dereferenced.
 
         Note: this could potentially be a smart factory, but currently it is not because
         we'd have to figure out when to expire the cache and that has some time.
         """
         if not self._json:
-            self._json = httpx.get(
+             # FIXME: this needs better error handling
+            r : httpx.Response = httpx.get(
                 self._uri,
                 follow_redirects=True,
                 verify=False,
                 headers={"Accept": "application/activity+json"})
+            if r.status_code != 200:
```

Comment:
FYI, `httpx.Response` has a `raise_for_status` method that will raise an exception on any HTTP error status code.